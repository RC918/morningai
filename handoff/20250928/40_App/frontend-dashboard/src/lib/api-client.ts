const API_BASE_URL = (import.meta as any).env?.VITE_API_BASE_URL || 'http://localhost:5001'

export const customFetch = async (url: string, options: RequestInit = {}) => {
  const fullUrl = url.startsWith('http') ? url : `${API_BASE_URL}${url}`
  
  const config: RequestInit = {
    headers: {
      'Content-Type': 'application/json',
      ...options.headers,
    },
    ...options,
  }

  const token = localStorage.getItem('auth_token')
  if (token) {
    config.headers = {
      ...config.headers,
      Authorization: `Bearer ${token}`,
    }
  }

  try {
    const response = await fetch(fullUrl, config)
    
    if (!response.ok) {
      const errorData = await response.json().catch(() => ({}))
      throw new Error(errorData.error?.message || `HTTP error! status: ${response.status}`)
    }
    
    return await response.json()
  } catch (error) {
    console.error(`API request failed: ${fullUrl}`, error)
    throw error
  }
}
