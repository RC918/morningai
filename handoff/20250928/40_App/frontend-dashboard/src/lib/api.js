import { tokenManager } from './auth/tokenManager'

const API_BASE_URL = import.meta.env.VITE_API_BASE_URL || 'http://localhost:5001'
const USE_MOCK = import.meta.env.VITE_USE_MOCK === 'true'

class ApiClient {
  constructor() {
    this.baseURL = API_BASE_URL
    this.useMock = USE_MOCK
  }

  async request(endpoint, options = {}) {
    const requestId = Math.random().toString(36).substr(2, 9)
    const url = `${this.baseURL}/api${endpoint}`
    const config = {
      headers: {
        'Content-Type': 'application/json',
        'X-Request-ID': requestId,
        ...options.headers,
      },
      ...options,
    }

    try {
      const token = await tokenManager.getValidToken()
      if (token) {
        config.headers.Authorization = `Bearer ${token}`
      }
    } catch (error) {
      console.warn('Failed to get valid token:', error.message)
    }

    try {
      const response = await fetch(url, config)
      if (!response.ok) {
        const errorData = await response.json().catch(() => ({}))
        const error = new Error(errorData.error?.message || `HTTP error! status: ${response.status}`)
        error.status = response.status
        error.requestId = requestId
        error.endpoint = endpoint
        
        console.error(`API Error [${requestId}]: ${endpoint} - ${error.message}`, {
          status: response.status,
          url,
          config
        })
        
        window.dispatchEvent(new CustomEvent('api-error', {
          detail: { endpoint, error: error.message, status: response.status, requestId }
        }))

        if (window.Sentry) {
          window.Sentry.captureException(error, {
            tags: { section: 'api_client', endpoint },
            extra: { requestId, status: response.status, url }
          })
        }

        throw error
      }
      return await response.json()
    } catch (error) {
      if (!error.requestId) {
        error.requestId = requestId
        error.endpoint = endpoint
        
        console.error(`Network Error [${requestId}]: ${endpoint} - ${error.message}`, {
          url,
          config,
          errorType: error.name
        })
        
        window.dispatchEvent(new CustomEvent('api-error', {
          detail: { endpoint, error: error.message, status: 0, requestId, type: 'network' }
        }))

        if (window.Sentry) {
          window.Sentry.captureException(error, {
            tags: { section: 'api_client', endpoint, error_type: 'network' },
            extra: { requestId, url }
          })
        }
      }
      
      throw error
    }
  }

  async checkHealth() {
    try {
      const response = await this.request('/health')
      return { healthy: true, ...response }
    } catch (error) {
      console.warn('Backend health check failed:', error.message)
      return { healthy: false, error: error.message }
    }
  }

  async requestWithRetry(endpoint, options = {}, maxRetries = 2) {
    let lastError
    for (let attempt = 0; attempt <= maxRetries; attempt++) {
      try {
        return await this.request(endpoint, options)
      } catch (error) {
        lastError = error
        if (attempt < maxRetries && (error.status === 0 || error.status >= 500)) {
          console.warn(`Retry ${attempt + 1}/${maxRetries} for ${endpoint}:`, error.message)
          await new Promise(resolve => setTimeout(resolve, 1000 * (attempt + 1)))
          continue
        }
        break
      }
    }
    throw lastError
  }

  async verifyAuth() {
    return this.request('/auth/verify')
  }

  async login(credentials) {
    const response = await this.request('/auth/login', {
      method: 'POST',
      body: JSON.stringify(credentials),
    })
    
    if (response.token && response.refresh_token) {
      tokenManager.setTokens(response.token, response.refresh_token, response.expires_in)
    }
    
    return response
  }

  async refreshToken(refreshToken) {
    return this.request('/auth/refresh', {
      method: 'POST',
      body: JSON.stringify({ refresh_token: refreshToken }),
    })
  }

  async ssoCallback(provider, code, codeVerifier, redirectUri) {
    const response = await this.request('/auth/sso/callback', {
      method: 'POST',
      body: JSON.stringify({
        provider,
        code,
        code_verifier: codeVerifier,
        redirect_uri: redirectUri
      }),
    })
    
    if (response.token && response.refresh_token) {
      tokenManager.setTokens(response.token, response.refresh_token, response.expires_in)
    }
    
    return response
  }

  async getBillingPlans() {
    const endpoint = this.useMock ? '/checkout/mock' : '/billing/plans'
    const data = await this.request(endpoint)
    return this.useMock ? data.pricing_tiers || [] : data.plans || []
  }

  async createCheckoutSession(sessionData) {
    const endpoint = this.useMock ? '/checkout/mock' : '/billing/checkout/session'
    return this.request(endpoint, {
      method: 'POST',
      body: JSON.stringify(sessionData),
    })
  }

  async getDashboardData() {
    return this.requestWithRetry('/dashboard/data')
  }

  async getDashboardWidgets() {
    return this.requestWithRetry('/dashboard/widgets')
  }

  async getReportTemplates() {
    return this.request('/reports/templates')
  }

  async getReportHistory() {
    return this.request('/reports/history')
  }

  async generateReport(reportData) {
    return this.request('/reports/generate', {
      method: 'POST',
      body: JSON.stringify(reportData),
    })
  }

  async getSettings() {
    return this.request('/settings')
  }

  async saveSettings(settings) {
    return this.request('/settings', {
      method: 'POST',
      body: JSON.stringify(settings),
    })
  }

  async get(endpoint) {
    return this.request(endpoint, {
      method: 'GET',
    })
  }

  async post(endpoint, data) {
    return this.request(endpoint, {
      method: 'POST',
      body: JSON.stringify(data),
    })
  }
}

export const apiClient = new ApiClient()
export default apiClient
