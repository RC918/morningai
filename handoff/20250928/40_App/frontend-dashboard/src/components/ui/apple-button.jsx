import * as React from "react"
import { Slot } from "@radix-ui/react-slot"
import { cva } from "class-variance-authority"
import { motion } from "framer-motion"

import { cn } from "@/lib/utils"
import { getSpringConfig, triggerHaptic } from "@/lib/spring-animation"

const appleButtonVariants = cva(
  "inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0",
  {
    variants: {
      variant: {
        primary:
          "bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm",
        secondary:
          "bg-secondary text-secondary-foreground shadow-sm hover:shadow-md focus-visible:ring-secondary/30 active:shadow-sm",
        destructive:
          "bg-destructive text-white shadow-sm hover:shadow-md focus-visible:ring-destructive/30 active:shadow-sm",
        outline:
          "border border-input bg-background/80 backdrop-blur-sm shadow-xs hover:bg-accent hover:text-accent-foreground focus-visible:ring-primary/20 active:bg-accent/80",
        ghost:
          "hover:bg-accent/80 hover:text-accent-foreground focus-visible:ring-primary/20 active:bg-accent",
        link:
          "text-primary underline-offset-4 hover:underline focus-visible:ring-primary/20",
        filled:
          "bg-accent text-accent-foreground shadow-xs hover:shadow-sm focus-visible:ring-accent/30 active:shadow-xs",
        tinted:
          "bg-primary/10 text-primary hover:bg-primary/20 focus-visible:ring-primary/30 active:bg-primary/15",
      },
      size: {
        sm: "h-8 rounded-lg gap-1.5 px-3 text-sm has-[>svg]:px-2.5",
        default: "h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3",
        lg: "h-12 rounded-xl gap-2 px-6 text-base has-[>svg]:px-4",
        icon: "size-10 rounded-xl",
        "icon-sm": "size-8 rounded-lg",
        "icon-lg": "size-12 rounded-xl",
      },
      haptic: {
        none: "",
        light: "",
        medium: "",
        heavy: "",
      },
    },
    defaultVariants: {
      variant: "primary",
      size: "default",
      haptic: "medium",
    },
  }
)

function AppleButton({
  className,
  variant,
  size,
  haptic = "medium",
  asChild = false,
  disabled = false,
  children,
  onClick,
  ...props
}) {
  const buttonRef = React.useRef(null)
  const Comp = asChild ? Slot : motion.button

  const handleClick = React.useCallback((e) => {
    if (disabled) return
    
    if (haptic !== "none" && buttonRef.current) {
      triggerHaptic(buttonRef.current, haptic)
    }
    
    onClick?.(e)
  }, [disabled, haptic, onClick])

  const springConfig = getSpringConfig('snappy')

  return (
    <Comp
      ref={buttonRef}
      data-slot="button"
      className={cn(appleButtonVariants({ variant, size, className }))}
      disabled={disabled}
      onClick={handleClick}
      whileHover={disabled ? {} : { scale: 1.02 }}
      whileTap={disabled ? {} : { scale: 0.98 }}
      transition={springConfig}
      {...props}
    >
      {children}
    </Comp>
  )
}

AppleButton.displayName = "AppleButton"

export { AppleButton, appleButtonVariants }
