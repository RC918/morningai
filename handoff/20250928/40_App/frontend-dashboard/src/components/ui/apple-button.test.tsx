/**
 * AppleButton Component Unit Tests
 * Target: 80% code coverage
 */

import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { AppleButton } from './apple-button';
import * as springAnimation from '@/lib/spring-animation';

vi.mock('@/lib/spring-animation', () => ({
  getSpringConfig: vi.fn(() => ({
    type: 'spring',
    stiffness: 300,
    damping: 30,
  })),
  triggerHaptic: vi.fn(),
}));

describe('AppleButton', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe('Rendering', () => {
    it('renders with default props', () => {
      render(<AppleButton>Click me</AppleButton>);
      const button = screen.getByRole('button', { name: /click me/i });
      
      expect(button).toBeInTheDocument();
      expect(button).toHaveTextContent('Click me');
    });

    it('renders with children', () => {
      render(
        <AppleButton>
          <span>Icon</span>
          <span>Text</span>
        </AppleButton>
      );
      
      expect(screen.getByText('Icon')).toBeInTheDocument();
      expect(screen.getByText('Text')).toBeInTheDocument();
    });

    it('applies data-slot attribute', () => {
      render(<AppleButton>Button</AppleButton>);
      const button = screen.getByRole('button');
      
      expect(button).toHaveAttribute('data-slot', 'button');
    });
  });

  describe('Variants', () => {
    const variants = [
      'primary',
      'secondary',
      'destructive',
      'outline',
      'ghost',
      'link',
      'filled',
      'tinted',
    ] as const;

    variants.forEach((variant) => {
      it(`renders ${variant} variant correctly`, () => {
        render(<AppleButton variant={variant}>{variant}</AppleButton>);
        const button = screen.getByRole('button');
        
        expect(button).toBeInTheDocument();
        expect(button).toHaveTextContent(variant);
      });
    });
  });

  describe('Sizes', () => {
    const sizes = ['sm', 'default', 'lg', 'icon', 'icon-sm', 'icon-lg'] as const;

    sizes.forEach((size) => {
      it(`renders ${size} size correctly`, () => {
        render(<AppleButton size={size}>Button</AppleButton>);
        const button = screen.getByRole('button');
        
        expect(button).toBeInTheDocument();
      });
    });
  });

  describe('Haptic Feedback', () => {
    it('triggers haptic feedback on click with default level', async () => {
      const user = userEvent.setup();
      render(<AppleButton>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(springAnimation.triggerHaptic).toHaveBeenCalledWith(
        expect.any(Object),
        'medium'
      );
    });

    it('triggers light haptic feedback', async () => {
      const user = userEvent.setup();
      render(<AppleButton haptic="light">Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(springAnimation.triggerHaptic).toHaveBeenCalledWith(
        expect.any(Object),
        'light'
      );
    });

    it('triggers heavy haptic feedback', async () => {
      const user = userEvent.setup();
      render(<AppleButton haptic="heavy">Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(springAnimation.triggerHaptic).toHaveBeenCalledWith(
        expect.any(Object),
        'heavy'
      );
    });

    it('does not trigger haptic feedback when haptic is none', async () => {
      const user = userEvent.setup();
      render(<AppleButton haptic="none">Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(springAnimation.triggerHaptic).not.toHaveBeenCalled();
    });

    it('does not trigger haptic feedback when disabled', async () => {
      const user = userEvent.setup();
      render(<AppleButton disabled>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(springAnimation.triggerHaptic).not.toHaveBeenCalled();
    });
  });

  describe('Click Handler', () => {
    it('calls onClick when clicked', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();
      render(<AppleButton onClick={handleClick}>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it('does not call onClick when disabled', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();
      render(<AppleButton onClick={handleClick} disabled>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(handleClick).not.toHaveBeenCalled();
    });

    it('passes event to onClick handler', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();
      render(<AppleButton onClick={handleClick}>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(handleClick).toHaveBeenCalledWith(expect.any(Object));
    });
  });

  describe('Disabled State', () => {
    it('renders disabled button', () => {
      render(<AppleButton disabled>Disabled</AppleButton>);
      const button = screen.getByRole('button');
      
      expect(button).toBeDisabled();
    });

    it('applies disabled attribute', () => {
      render(<AppleButton disabled>Disabled</AppleButton>);
      const button = screen.getByRole('button');
      
      expect(button).toHaveAttribute('disabled');
    });

    it('does not trigger haptic feedback when disabled', async () => {
      const user = userEvent.setup();
      render(<AppleButton disabled>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(springAnimation.triggerHaptic).not.toHaveBeenCalled();
    });
  });

  describe('Custom className', () => {
    it('applies custom className', () => {
      render(<AppleButton className="custom-class">Button</AppleButton>);
      const button = screen.getByRole('button');
      
      expect(button).toHaveClass('custom-class');
    });

    it('merges custom className with variant classes', () => {
      render(
        <AppleButton variant="primary" className="custom-class">
          Button
        </AppleButton>
      );
      const button = screen.getByRole('button');
      
      expect(button).toHaveClass('custom-class');
    });
  });

  describe('Spring Animation Integration', () => {
    it('calls getSpringConfig on render', () => {
      render(<AppleButton>Button</AppleButton>);
      
      expect(springAnimation.getSpringConfig).toHaveBeenCalledWith('snappy');
    });

    it('uses spring config for animations', () => {
      const mockConfig = {
        type: 'spring',
        stiffness: 300,
        damping: 30,
      };
      vi.mocked(springAnimation.getSpringConfig).mockReturnValue(mockConfig);
      
      render(<AppleButton>Button</AppleButton>);
      
      expect(springAnimation.getSpringConfig).toHaveBeenCalled();
    });
  });

  describe('Accessibility', () => {
    it('is keyboard accessible', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();
      render(<AppleButton onClick={handleClick}>Button</AppleButton>);
      const button = screen.getByRole('button');
      
      button.focus();
      expect(button).toHaveFocus();
      
      await user.keyboard('{Enter}');
      expect(handleClick).toHaveBeenCalled();
    });

    it('supports Space key activation', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();
      render(<AppleButton onClick={handleClick}>Button</AppleButton>);
      const button = screen.getByRole('button');
      
      button.focus();
      await user.keyboard(' ');
      
      expect(handleClick).toHaveBeenCalled();
    });

    it('has correct role', () => {
      render(<AppleButton>Button</AppleButton>);
      const button = screen.getByRole('button');
      
      expect(button).toBeInTheDocument();
    });

    it('supports aria-label', () => {
      render(<AppleButton aria-label="Custom label">Button</AppleButton>);
      const button = screen.getByRole('button', { name: /custom label/i });
      
      expect(button).toBeInTheDocument();
    });

    it('supports aria-describedby', () => {
      render(
        <>
          <AppleButton aria-describedby="description">Button</AppleButton>
          <div id="description">Description text</div>
        </>
      );
      const button = screen.getByRole('button');
      
      expect(button).toHaveAttribute('aria-describedby', 'description');
    });
  });

  describe('asChild prop', () => {
    it('renders as Slot when asChild is true', () => {
      render(
        <AppleButton asChild>
          <a href="/test">Link Button</a>
        </AppleButton>
      );
      
      const link = screen.getByRole('link');
      expect(link).toBeInTheDocument();
      expect(link).toHaveAttribute('href', '/test');
    });
  });

  describe('Variant Combinations', () => {
    it('renders primary button with small size', () => {
      render(
        <AppleButton variant="primary" size="sm">
          Small Primary
        </AppleButton>
      );
      const button = screen.getByRole('button');
      
      expect(button).toBeInTheDocument();
      expect(button).toHaveTextContent('Small Primary');
    });

    it('renders destructive button with large size', () => {
      render(
        <AppleButton variant="destructive" size="lg">
          Large Destructive
        </AppleButton>
      );
      const button = screen.getByRole('button');
      
      expect(button).toBeInTheDocument();
      expect(button).toHaveTextContent('Large Destructive');
    });

    it('renders outline button with icon size', () => {
      render(
        <AppleButton variant="outline" size="icon">
          +
        </AppleButton>
      );
      const button = screen.getByRole('button');
      
      expect(button).toBeInTheDocument();
    });
  });

  describe('Edge Cases', () => {
    it('handles rapid clicks', async () => {
      const handleClick = vi.fn();
      const user = userEvent.setup();
      render(<AppleButton onClick={handleClick}>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      await user.click(button);
      await user.click(button);
      
      expect(handleClick).toHaveBeenCalledTimes(3);
      expect(springAnimation.triggerHaptic).toHaveBeenCalledTimes(3);
    });

    it('handles empty children', () => {
      render(<AppleButton />);
      const button = screen.getByRole('button');
      
      expect(button).toBeInTheDocument();
    });

    it('handles null onClick', async () => {
      const user = userEvent.setup();
      render(<AppleButton onClick={undefined}>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      
      expect(springAnimation.triggerHaptic).toHaveBeenCalled();
    });

    it('maintains ref through clicks', async () => {
      const user = userEvent.setup();
      render(<AppleButton>Click</AppleButton>);
      const button = screen.getByRole('button');
      
      await user.click(button);
      await user.click(button);
      
      expect(springAnimation.triggerHaptic).toHaveBeenCalledTimes(2);
      expect(springAnimation.triggerHaptic).toHaveBeenCalledWith(
        expect.any(Object),
        'medium'
      );
    });
  });

  describe('Snapshot Tests', () => {
    it('matches snapshot for primary variant', () => {
      const { container } = render(
        <AppleButton variant="primary">Primary</AppleButton>
      );
      
      expect(container).toMatchSnapshot();
    });

    it('matches snapshot for all variants', () => {
      const { container } = render(
        <div>
          <AppleButton variant="primary">Primary</AppleButton>
          <AppleButton variant="secondary">Secondary</AppleButton>
          <AppleButton variant="destructive">Destructive</AppleButton>
          <AppleButton variant="outline">Outline</AppleButton>
          <AppleButton variant="ghost">Ghost</AppleButton>
          <AppleButton variant="link">Link</AppleButton>
          <AppleButton variant="filled">Filled</AppleButton>
          <AppleButton variant="tinted">Tinted</AppleButton>
        </div>
      );
      
      expect(container).toMatchSnapshot();
    });

    it('matches snapshot for all sizes', () => {
      const { container } = render(
        <div>
          <AppleButton size="sm">Small</AppleButton>
          <AppleButton size="default">Default</AppleButton>
          <AppleButton size="lg">Large</AppleButton>
          <AppleButton size="icon">+</AppleButton>
          <AppleButton size="icon-sm">+</AppleButton>
          <AppleButton size="icon-lg">+</AppleButton>
        </div>
      );
      
      expect(container).toMatchSnapshot();
    });

    it('matches snapshot for disabled state', () => {
      const { container } = render(
        <div>
          <AppleButton>Normal</AppleButton>
          <AppleButton disabled>Disabled</AppleButton>
        </div>
      );
      
      expect(container).toMatchSnapshot();
    });
  });
});
