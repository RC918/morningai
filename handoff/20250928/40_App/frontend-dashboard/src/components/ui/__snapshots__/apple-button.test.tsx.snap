// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppleButton > Snapshot Tests > matches snapshot for all sizes 1`] = `
<div>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm h-8 rounded-lg gap-1.5 px-3 text-sm has-[>svg]:px-2.5"
      data-slot="button"
      tabindex="0"
    >
      Small
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Default
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm h-12 rounded-xl gap-2 px-6 text-base has-[>svg]:px-4"
      data-slot="button"
      tabindex="0"
    >
      Large
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm size-10 rounded-xl"
      data-slot="button"
      tabindex="0"
    >
      +
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm size-8 rounded-lg"
      data-slot="button"
      tabindex="0"
    >
      +
    </button>
    <button
      class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm size-12 rounded-xl"
      data-slot="button"
      tabindex="0"
    >
      +
    </button>
  </div>
</div>
`;

exports[`AppleButton > Snapshot Tests > matches snapshot for all variants 1`] = `
<div>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Primary
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-secondary text-secondary-foreground shadow-sm hover:shadow-md focus-visible:ring-secondary/30 active:shadow-sm h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Secondary
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-destructive text-white shadow-sm hover:shadow-md focus-visible:ring-destructive/30 active:shadow-sm h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Destructive
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 border border-input bg-background/80 backdrop-blur-sm shadow-xs hover:bg-accent hover:text-accent-foreground focus-visible:ring-primary/20 active:bg-accent/80 h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Outline
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 hover:bg-accent/80 hover:text-accent-foreground focus-visible:ring-primary/20 active:bg-accent h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Ghost
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 text-primary underline-offset-4 hover:underline focus-visible:ring-primary/20 h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Link
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-accent text-accent-foreground shadow-xs hover:shadow-sm focus-visible:ring-accent/30 active:shadow-xs h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Filled
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary/10 text-primary hover:bg-primary/20 focus-visible:ring-primary/30 active:bg-primary/15 h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Tinted
    </button>
  </div>
</div>
`;

exports[`AppleButton > Snapshot Tests > matches snapshot for disabled state 1`] = `
<div>
  <div>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      tabindex="0"
    >
      Normal
    </button>
    <button
      class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
      data-slot="button"
      disabled=""
      tabindex="0"
    >
      Disabled
    </button>
  </div>
</div>
`;

exports[`AppleButton > Snapshot Tests > matches snapshot for primary variant 1`] = `
<div>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:ring-[3px] focus-visible:ring-offset-0 bg-primary text-primary-foreground shadow-sm hover:shadow-md focus-visible:ring-primary/30 active:shadow-sm h-10 rounded-xl gap-2 px-4 text-sm has-[>svg]:px-3"
    data-slot="button"
    tabindex="0"
  >
    Primary
  </button>
</div>
`;
